frontend/public/components/environment.jsx

```diff
@@ -94,12 +101,17 @@ export const EnvironmentPage = connect(stateToProps)(
       this.saveChanges = this._saveChanges.bind(this);
       this.updateEnvVars = this._updateEnvVars.bind(this);
       this.selectContainer = this._selectContainer.bind(this);
+      const isContainerArray = _.isArray(this.props.rawEnvData.containers);
 
-      const currentEnvVars = envVarsToArray(this.props.rawEnvData);
+      const currentEnvVars = {};
+      currentEnvVars.containers = envVarsToArray(getVariablesFromObject(this.props.rawEnvData, isContainerArray));
```

Comment:
So we're always setting `currentEnvVars.containers` even if it's not a container array? That's confusing. Is there another way?